daemon/api_validate_test.go

```diff
@@ -542,6 +564,44 @@ func (s *apiValidationSetsSuite) TestGetValidationSetSpecificSequenceFromRemote(
 	})
 }
 
+func (s *apiValidationSetsSuite) TestGetValidationSetFromRemoteFallbackToLocalAssertion(c *check.C) {
+	s.mockSeqFormingAssertionFn = func(assertType *asserts.AssertionType, sequenceKey []string, sequence int, user *auth.UserState) (asserts.Assertion, error) {
+		// not found in the store
+		return nil, &asserts.NotFoundError{
+			Type: assertType,
+		}
+	}
+	restore := daemon.MockCheckInstalledSnaps(func(vsets *snapasserts.ValidationSets, snaps []*snapasserts.InstalledSnap) error {
+		// nil indicates successful validation
+		return nil
+	})
+	defer restore()
+
+	st := s.d.Overlord().State()
+	st.Lock()
+	c.Assert(assertstate.Add(st, s.dev1acct), check.IsNil)
+	c.Assert(assertstate.Add(st, s.acct1Key), check.IsNil)
+	// assertion available in the local db (from snap ack)
+	vs := s.mockAssertByDev1(c, "bar", "2")
+	c.Assert(assertstate.Add(st, vs), check.IsNil)
```

Comment:
the various Add here could use  assertstatetest.AddMany

Reply:
I'll address it in the followup along with some other cleanups of these tests.